src/Coordination/KeeperStateMachine.cpp

```diff
@@ -308,7 +311,25 @@ int KeeperStateMachine::read_logical_snp_obj(
                             s.get_last_log_idx(), latest_snapshot_meta->get_last_log_idx());
             return -1;
         }
-        data_out = nuraft::buffer::clone(*latest_snapshot_buf);
+        int fd = ::open(latest_snapshot_path.c_str(), O_RDONLY);
+        LOG_INFO(log, "Opening file {} for read_logical_snp_obj", latest_snapshot_path);
+        if (fd < 0)
+        {
+            LOG_WARNING(log, "Error opening {}.", latest_snapshot_path);
+            return -1;
+        }
+        auto file_size = ::lseek(fd, 0, SEEK_END);
```

Comment:
Maybe we can use this code https://github.com/ClickHouse/ClickHouse/blob/e1fb220bb721a5db8b125c6768aa905d637084fe/src/Common/getMappedArea.h?

Reply:
This is used only on Linux, not suitable for mac os x.